smalltalksrc/VMMaker/VMRememberedSet.class.st

```diff
@@ -164,6 +164,15 @@ VMRememberedSet >> initialize [
 VMRememberedSet >> initializeRememberedSetShouldStartEmpty: shouldStartEmpty [
 	| obj |
 	obj := self objectOop.
+	
+	(manager isInOldSpace: obj) 
+		ifFalse: [ 
+			manager logWarn: 'Remembered Set is in an invalid position %p. Signal of a corrupted root table.' _: obj.
+			shouldStartEmpty 
+				ifTrue: [ 
+					obj := manager nilObject.
+					manager logWarn: 'As it starts empty, we can ignore it' ]]. 
```

Comment:
If it should not start empty, there is nothing to do and eventually will crash, right?

Reply:
Yes, maybe it needs to just hard quit?